independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/observers/inheritance/generated/GeneratedObserverClassNotSharedTest.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.arc.test.observers.inheritance.generated;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ClientProxy;
+import io.quarkus.arc.test.ArcTestContainer;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Event;
+import javax.enterprise.event.Observes;
+import javax.inject.Singleton;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class GeneratedObserverClassNotSharedTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = new ArcTestContainer(Alpha.class, Bravo.class);
+
+    static final Set<Alpha> ALPHAS = new CopyOnWriteArraySet<>();
+
+    @Test
+    public void testIdentities() {
+        ALPHAS.clear();
+        Event<Alpha> event = Arc.container().beanManager().getEvent().select(Alpha.class);
+        // Fire two events and verify that only one instance of Alpha and Bravo was used
+        event.fire(new Alpha());
+        event.fire(new Alpha());
```

Comment:
Did you want to fire `Bravo` too?

Reply:
Ah, I probably misunderstood what you're doing here :-) The fact that you're firing 2 events and then asserting `ALPHAS` has 2 elements confused me, I guess. Those twos are two different twos :laughing:

Reply:
Is there any reason why you are throwing Alphas? It makes the test a bit confusing.

Reply:
The event payload is irrelevant. I used `Alpha` so that no other class is needed for the payload. I'll change that to make the test more understandable.

Reply:
I hope it's better now ;-)